openmdao/core/system.py

```diff
@@ -1969,9 +1980,19 @@ def _setup_recording(self):
                     myresiduals = myoutputs
 
             elif options['record_residuals']:
-                abs2prom = self._var_abs2prom['output']
+                match_names = match_names | set(self._residuals.keys())
                 myresiduals = [n for n in self._residuals._abs_iter()
-                               if check_path(abs2prom[n], incl, excl)]
+                               if check_path(abs2prom_output[n], incl, excl)]
+
+            # check that all exclude/include globs have at least one matching output or input name
+            for pattern in excl:
+                if not has_match(pattern, match_names):
```

Comment:
It's kind of a bummer that we're doing the glob pattern match a second time for every variable just to check for a bad pattern. Could we improve things a little by only checking the `excl` patterns against the vars that we did match and only checking the 'incl' patterns against vars that we didn't match? I guess that could miss cases where you had multiple patterns where an earlier one matched and a later one didn't, but would the user even care in that case since the intended set of vars was matched even though certain patterns didn't match anything?